controller/volume_controller.go

```diff
@@ -1400,6 +1400,7 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 				if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" {
 					r.Spec.FailedAt = vc.nowHandler()
 					r.Spec.DesireState = longhorn.InstanceStateStopped
+					r.Spec.RebuildRetryCount = scheduler.FailedReplicaMaxRetryCount
```

Comment:
The reused failed replicas will have a similar status `r.Spec.HealthyAt == "" && r.Spec.FailedAt == ""` as well... In other words, this PR will clean up these replicas
https://github.com/longhorn/longhorn-manager/blob/cbcd5c7beab27237f2761b3f74d8fb4167e981da/controller/volume_controller.go#L1721-L1723

Reply:
Check if disk is empty or nodeID is empty then set `RebuildRetryCount` to clean up
@shuo-wu WDYT?

Reply:
> After commit 15f850c, although node 2 scheduling enabled, the 2 failed replicas in previous step remain unscheduled (N/A), and will not be reused to restore data, the volume has to create a new replica to schedule to node 2, so the volume will have 2 scheduled replicas and 2 failed replicas from the previous step, so assertion failed.

The real issue seems how to reuse the failed replicas rather than delete failed replicas. Right?

Reply:
These failed replicas is not used before, so I think it could be deleted and wait for replenishing when volume attached.
Added `check if NodeID or DiskId is empty`

Reply:
Agree with Derek's comment.
You can consider the following besides the current fix:
1. The scheduler reusage should ignore the unscheduled failed replicas.
2. The unscheduled failed replicas should be directly considered as corrupted and stale replicas.

Reply:
>1. The scheduler reusage should ignore the unscheduled failed replicas.

Do you mean `reused` ? We did `ignore` unscheduled failed replicas (not reused and not deleted) now.

>2. The unscheduled failed replicas should be directly considered as corrupted and stale replicas.

Unscheduled replicas would be marked as failed when volume is detached and set `RebuildRetryCount` would make unscheduled failed replicas considered as corrupted and stale replicas.

Reply:
Yes. I just checked the code. This fix looks good enough.